### PR TITLE
Meta Tags for Course, Schedule, Department and Explore pages

### PR DESCRIPTION
**Pull request by RichtXO**
SEO is almost done, just need to fix the HTML page with JS disabled

**Issue**

Improving Lighthouse score

### CODE REVIEW

**Comment by RichtXO** [comment]
@jshom Got 100 in SEO 👍 
But lmk if that's how to render the course page! I switched back using vue-meta because it would just easier to code lol

File: src/web/nginx.conf [hunk 1]
```diff
@@ -13,17 +13,25 @@ http {
   default_type application/octet-stream;
   keepalive_timeout  65;
 
-
   server{
     listen 80;
     listen [::]:80;
     server_name ${HOST};
-    return 301 https://${HOST};
+    return 301 https://server_name$request_uri;
```
Comment (jshom) [inline src/web/nginx.conf:20]: lets do  ${HOST}$request_uri. might solve temp deploy i think as well

**Comment by jshom** [comment]
This is awesome! Small tweaks and should be good to merge.
1. We need meta keywords for course pages: For each page we need, `RPI (we'll push this into admin panel later), YACS [then course specific tags]: COGS, 2120, COGS-2120, INTRO TO COGNITIVE SCIENCE
2. Same regarding keywords, for department pages, should be the same except without the course title, level and name ofcourse.
3. Title for the COGS department page should be COGS | YACS, right now it is just YACS
4. Explore title, should be Explore | YACS

Besides that lgtm, we'll be adding more in the meta fields later, but I think this is an awesome start!

**Review by jshom — APPROVED** [review]

File: src/web/src/pages/CoursePage.vue [hunk 2]
```diff
@@ -117,5 +117,18 @@ export default {
...
     },
   },
+  metaInfo() {
+    return {
+      title: this.courseTitle,
+      titleTemplate: "%s | YACS",
+      meta: [
+        { charset: 'utf-8' },
+        { name: 'description', content: this.courseObj.description},
+        { property: 'og:title', content: "YACS - RPI Course Scheduler"},
```
Comment (jshom) [inline src/web/src/pages/CoursePage.vue:127]: Also, maybe lets switch this to RPI - YACS Course Scheduler, we'll parametrize this a bit later in admin panel

**Review by jshom — APPROVED** [review]

File: src/web/src/main.js [hunk 3]
```diff
@@ -2,21 +2,26 @@ import "@babel/polyfill";
...
 import router from "./routes";
 import VueCookies from "vue-cookies";
-import "vue-virtual-scroller/dist/vue-virtual-scroller.css";
+import Meta from 'vue-meta';
 
 Vue.config.productionTip = false;
 
 Vue.use(VueRouter);
 Vue.use(VueCookies);
+Vue.use(Meta);
```
Comment (123joshuawu) [inline src/web/src/main.js:18]: looks like `Meta` being used twice oops

File: src/web/src/pages/CoursePage.vue [hunk 4]
```diff
@@ -117,5 +117,19 @@ export default {
       return precoreqtext;
     },
   },
+  metaInfo() {
+    return {
+      title: this.courseTitle,
```
Comment (123joshuawu) [inline src/web/src/pages/CoursePage.vue:122]: can maybe extract some of this to a function to avoid reduce redundancy?

**Review by 123joshuawu — COMMENTED** [review]
lgtm

File: src/web/src/main.js (hunk 3 above)
Comment (RichtXO) [reply src/web/src/main.js:18]: oops, corrected now!

**Comment by jshom** [comment]
@RichtXO can you take a look at the merge conflicts when you get a chance, thx!

**Review by 123joshuawu — COMMENTED** [review]
seems like `package.json` is missing a dependency `vue-meta` otherwise looks good!

**Comment by 123joshuawu** [comment]
I think it's in the wrong package.json, we should probably delete the root level package.json at some point

**Comment by RichtXO** [comment]
> I think it's in the wrong package.json, we should probably delete the root level package.json at some point. I've gotten them mixed up plenty of times

@123joshuawu LMAO yep! However, kinda a new problem... I can't get the compression-webpack to work... `tapPromise` seems to be EOL

**Comment by 123joshuawu** [comment]
Hmm, we'll probably just have to revert to an older version or find a new package. Up to you

**Review by 123joshuawu — APPROVED** [review]

**Comment by RichtXO** [comment]
Lol, took me a while to know what I was supposed to do. Not going to make a function, but have a parent-child thing crap. Should be able to get it tonight and get this PR out!

**Pull request merged**